### PR TITLE
daily udate

### PR DESCRIPTION
**Pull request by leaxoy**
There is no new feature add. But does some others.
1. update `typescript` to 2.0
2. use `npm install @types/{package}` instead of put `typing file` to `typings/` but for package `json-rpc2` there is no `typing file` exists. Do you have any idea on this? @ramya-rao-a

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,42 +11,45 @@
     *  Fix Generate Test for Current function when the function is a method on a type. [PR 657](https://github.com/Microsoft/vscode-go/pull/657)
 * [Potter Dai (@PotterDai)](https://github.com/PotterDai)
     *  Fix Find all References when using multiple GOPATH where one is the substring of the other. [PR 658](https://github.com/Microsoft/vscode-go/pull/658)
-* [Ramya Rao (@ramya-rao-a)](https://github.com/ramya-rao-a)
```
Comment (ramya-rao-a) [inline CHANGELOG.md]: Can you undo the changes to this file?

File: package.json [hunk 2]
```diff
@@ -39,10 +39,16 @@
     "vscode-debugprotocol": "^1.11.0"
   },
   "devDependencies": {
-    "fs-extra": "^0.30.0",
-    "tslint": "^3.14.0",
-    "typescript": "1.8.10",
-    "vscode": "^0.11.17"
+    "@types/diff": "0.0.31",
+    "@types/fs-extra": "0.0.35",
+    "@types/lodash": "^4.14.40",
```
Comment (ramya-rao-a) [inline package.json]: The travis tests for linux are failing to find types/lodash. Am not sure why, but for now you can remove lodash from your refactoring

**Comment by ramya-rao-a** [comment]
@roblourens You have any idea on @leaxoy's question on `json-rpc2` ?

File: CHANGELOG.md (hunk 1 above)
Comment (leaxoy) [inline CHANGELOG.md]: Sorry, maybe a mistake when merge comment!😂

File: package.json (hunk 2 above)
Comment (leaxoy) [inline package.json]: I notice that travis use an old node v0.10.36 with an old npm v1.4.28. Is this affected? Can we upgrade it to recent version?

**Comment by roblourens** [comment]
Our json-rpc2 typings are not official and not published anywhere. @leaxoy you could try to publish it to DefinitelyTyped if you want to get it from @types, but I have no idea if it's complete or high quality enough to do that.

Interestingly, this repo is the first result if you google "json-rpc2 typings".

File: package.json (hunk 2 above)
Comment (ramya-rao-a) [inline package.json]: Yes, sure. http://austinpray.com/ops/2015/09/20/change-travis-node-version.html talks about how we can do that. And here is an example that does the same https://github.com/nodejs/nan/blob/master/.travis.yml

**Comment by ramya-rao-a** [comment]
@leaxoy Do you need any help with this?

**Comment by ramya-rao-a** [comment]
@leaxoy For using the text mate snippet syntax, I needed the latest vscode apis which in turn needed the changes in this PR. Therefore, I am closing this PR in favor of #709 
Thanks for your work on this, I look forward to more.
